mapie/conformity_scores/classification.py

```diff
@@ -0,0 +1,224 @@
+            uncertainty of the confidence interval.
+
+        estimator: EnsembleClassifier
+            Estimator that is fitted to predict y from X.
+
+        Returns:
+        --------
+        NDArray
+            Array of predictions.
+        """
+
+    @abstractmethod
+    def get_conformity_score_quantiles(
+        self,
+        conformity_scores: NDArray,
+        alpha_np: NDArray,
+        estimator: EnsembleClassifier,
+        **kwargs
+    ) -> NDArray:
+        """
+        Abstract method to get quantiles of the conformity scores.
+
+        This method should be implemented by any subclass of the current class.
+
+        Parameters:
+        -----------
+        conformity_scores: NDArray of shape (n_samples,)
+            Conformity scores for each sample.
+
+        alpha_np: NDArray of shape (n_alpha,)
+            NDArray of floats between 0 and 1, representing the uncertainty
+            of the confidence interval.
+
+        estimator: EnsembleClassifier
+            Estimator that is fitted to predict y from X.
+
+        Returns:
+        --------
+        NDArray
+            Array of quantiles with respect to alpha_np.
+        """
+
+    @abstractmethod
+    def get_prediction_sets(
+        self,
+        y_pred_proba: NDArray,
+        conformity_scores: NDArray,
+        alpha_np: NDArray,
+        estimator: EnsembleClassifier,
+        **kwargs
+    ) -> NDArray:
+        """
+        Abstract method to generate prediction sets based on the probability
+        predictions, the conformity scores and the uncertainty level.
+
+        This method should be implemented by any subclass of the current class.
+
+        Parameters:
+        -----------
+        y_pred_proba: NDArray of shape (n_samples, n_classes)
+            Target prediction.
+
+        conformity_scores: NDArray of shape (n_samples,)
+            Conformity scores for each sample.
+
+        alpha_np: NDArray of shape (n_alpha,)
+            NDArray of floats between 0 and 1, representing the uncertainty
+            of the confidence interval.
+
+        estimator: EnsembleClassifier
+            Estimator that is fitted to predict y from X.
+
+        Returns:
+        --------
+        NDArray
+            Array of quantiles with respect to alpha_np.
+        """
+
+    def get_sets(
+        self,
+        X: NDArray,
+        alpha_np: NDArray,
+        estimator: EnsembleClassifier,
+        conformity_scores: NDArray,
+        **kwargs
+    ) -> NDArray:
+        """
+        Compute classes of the prediction sets from the observed values,
+        the estimator of type ``EnsembleClassifier`` and the conformity scores.
+
+        Parameters
+        ----------
+        X: NDArray of shape (n_samples, n_features)
+            Observed feature values.
+
+        alpha_np: NDArray of shape (n_alpha,)
+            NDArray of floats between 0 and 1, representing the uncertainty
+            of the confidence interval.
```

Comment:
What do you mean by "confidence interval" here ?

Reply:
It should be replaced by "confidence set" 👍